polaris-react/src/components/Box/Box.tsx

```diff
@@ -270,7 +273,10 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
       '--pc-box-max-width': maxWidth,
       '--pc-box-outline-color': outlineColor
         ? `var(--p-color-${outlineColor})`
-        : undefined,
+        : 'transparent',
+      '--pc-box-outline-offset': outlineOffset?.startsWith('-')
+        ? `calc(-1 * var(--p-border-width-${outlineOffset.slice(1)}))`
+        : `var(--p-border-width-${outlineOffset})`,
```

Comment:
Heads up, we missed the `undefined` case when jamming on this solution e.g. the above would apply `var(--p-border-width-undefined)` as an inline style